src/Xamarin.Android.Build.Tasks/Generator/Generator.cs

```diff
@@ -60,15 +61,19 @@ public static bool CreateJavaSources (TaskLoggingHelper log, IEnumerable<TypeDef
 			return ok;
 		}
 
-		static string GetMonoInitSource (bool useSharedRuntime)
+		static string GetMonoInitSource (string androidSdkPlatform, bool useSharedRuntime)
 		{
 			// Lookup the mono init section from MonoRuntimeProvider:
 			// Mono Runtime Initialization {{{
 			// }}}
 			var builder = new StringBuilder ();
-			var suffix = useSharedRuntime ? "Shared" : "Bundled";
+			var runtime = useSharedRuntime ? "Shared" : "Bundled";
+			var api = "";
+			if (int.TryParse (androidSdkPlatform, out int apiLevel) && apiLevel < 21) {
```

Comment:
This might break when using the shared runtime since `MonoRuntimeProvider.Shared.20.java` currently doesn't exist. You probably already know that , however I thought I'd make the comment so we do not forget.